static/js/settings_user_groups.js

```diff
@@ -76,40 +75,43 @@ exports.populate_user_groups = function () {
         }
 
         var input = pill_container.children('.input');
-
-        input.typeahead({
-            items: 5,
-            fixed: true,
-            dropup: true,
-            source: function () {
+        (function set_up_typeahead() {
+            input.typeahead({
+                items: 5,
+                fixed: true,
+                dropup: true,
+                source: function () {
                 return user_pill.typeahead_source(pills);
-            },
-            highlighter: function (item) {
-                return typeahead_helper.render_person(item);
-            },
-            matcher: function (item) {
-                var query = this.query.toLowerCase();
-                return (item.email.toLowerCase().indexOf(query) !== -1
-                        || item.full_name.toLowerCase().indexOf(query) !== -1);
-            },
-            sorter: function (matches) {
-                return typeahead_helper.sort_recipientbox_typeahead(
-                    this.query, matches, "");
-            },
-            updater: function (user) {
-                append_user(user);
-                update_save_state();
-            },
-            stopAdvance: true,
-        });
+                },
+                highlighter: function (item) {
+                    return typeahead_helper.render_person(item);
+                },
+                matcher: function (item) {
+                    var query = this.query.toLowerCase();
+                    return (item.email.toLowerCase().indexOf(query) !== -1
+                            || item.full_name.toLowerCase().indexOf(query) !== -1);
+                },
+                sorter: function (matches) {
+                    return typeahead_helper.sort_recipientbox_typeahead(
+                        this.query, matches, "");
+                },
+                updater: function (user) {
+                    append_user(user);
+                    update_save_state();
+                },
+                stopAdvance: true,
+            });
+        }());
```

Comment:
I'm not sure this improves the codebase; this whole block is already wrapped in a function, and none of these define new variables.

Reply:
https://github.com/zulip/zulip/pull/8365#issuecomment-372025849, Steve and I thought this, and also we thought that it would be good to some early exit conditions, also see this -https://chat.zulip.org/#narrow/stream/91-code-review/subject/PR.20.238365 :)
sorry for the delay, I'm out of city right now.